keps/sig-windows/4802-windows-node-shutdown/README.md

```diff
@@ -0,0 +1,539 @@
+
+There’s no one way to make all types of applications (console/GUI and Windows services) detect a node shutdown, different types of applications differ 
+in their syntax and functionality.
+
+For general Windows console application, it can process the "CTRL_SHUTDOWN_EVENT", but according to [HandlerRoutine callback function](https://learn.microsoft.com/en-us/windows/console/handlerroutine#parameters)
+and the [SetConsoleCtrlHandler function](https://learn.microsoft.com/windows/console/setconsolectrlhandler?redirectedfrom=MSDN) 
+`Console functions, or any C run-time functions that call console functions, may not work reliably during processing of any of the three signals mentioned previously. 
+The reason is that some or all of the internal console cleanup routines may have been called before executing the process signal handler`.
+
+While running as a Windows service, programs will have more freedom to register a service control handler to monitoring various shutdown related events 
+such as "SERVICE_CONTROL_PRESHUTDOWN", "SERVICE_CONTROL_SHUTDOWN". So in order to configure and process the (pre)shutdown event from Windows system, 
+register the kubelet application as a Windows service program will be the most straightforward way.
+
+Also from the same document [SetConsoleCtrlHandler function](https://learn.microsoft.com/windows/console/setconsolectrlhandler?redirectedfrom=MSDN),
+It appears that "SERVICE_CONTROL_PRESHUTDOWN" is more flexible than "SERVICE_CONTROL_SHUTDOWN" for a Windows service to monitor and control
+
+### Implementation
+Similar with the linux case, "nodeshutdown_manager_windows.go" will be added to replace the current
+fake/empty implementation for Windows, it will act as the main role in the node shutdown scenario and go through similar 
+initialization/start process during kublet start up.
+
+Also, we will update the Windows service package to "AcceptPreShutdown" and add a pre-shutdown handler to process
+the "SERVICE_CONTROL_PRESHUTDOWN" event. This pre-shutdown handler will have an interface "ProcessShutdownEvent",
+which will be implemented by the "nodeshutdown_manager_windows"
+
+The entry point of running the kublet as a Windows service is "InitService", in which, we can initialize and register 
+the service control handler, the above "pre-shutdown handler" is part of this service control handler.
+
+At this "InitService" stage, the kubelet has not read the kubelet configs/flags yet, so the "nodeshutdown_manager_windows", 
+has not been instantiated, we will then set the "preshutdownhander" to be NIL, so will ignore the preshutdown event if any. 
+This should be safe as the kubelet itself has not been started, let alone the pods it manages, so there is no 
+need to do anything for shutting down at this stage.
+
+When kubelet proceeds with starting up and initialize the "nodeshutdown_manager_window", it will set the service control ehandler’s 
```

Comment:
```suggestion
When kubelet proceeds with starting up and initialize the "nodeshutdown_manager_window", it will set the service control handler’s 
```

Reply:
updated